src/Fixer/Alias/ArrayPushFixer.php

```diff
@@ -0,0 +1,211 @@
+    {
+        return $tokens->count() > 7 && $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $functionsAnalyzer = new FunctionsAnalyzer();
+
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->equals([T_STRING, 'array_push'])) {
+                continue;
+            }
+
+            if (!$functionsAnalyzer->isGlobalFunctionCall($tokens, $index)) {
+                continue; // redeclare/override
+            }
+
+            // meaningful before must be `<?php`, `}` or `;`
+
+            $callIndex = $index;
+            $index = $tokens->getPrevMeaningfulToken($index);
+            $namespaceSeparatorIndex = null;
+
+            if ($tokens[$index]->isGivenKind([T_NS_SEPARATOR])) {
+                $namespaceSeparatorIndex = $index;
+                $index = $tokens->getPrevMeaningfulToken($index);
+            }
+
+            if (!$tokens[$index]->equalsAny([';', '}', [T_OPEN_TAG]])) {
+                continue;
+            }
+
+            // figure out where the arguments list opens
+
+            $openBraceIndex = $tokens->getNextMeaningfulToken($callIndex);
+            $blockType = Tokens::detectBlockType($tokens[$openBraceIndex]);
+
+            if (null === $blockType || Tokens::BLOCK_TYPE_PARENTHESIS_BRACE !== $blockType['type']) {
+                continue;
+            }
+
+            // fast check if the first argument is a variable of some sort
+
+            $firstArgumentStart = $tokens->getNextMeaningfulToken($openBraceIndex);
+
+            if (!$tokens[$firstArgumentStart]->isGivenKind(T_VARIABLE)) {
+                continue;
+            }
+
+            // figure out where the arguments list closes
+
+            $closeBraceIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $openBraceIndex);
+
+            // meaningful after `)` must be `;`, `?>` or nothing
+
+            $afterCloseBraceIndex = $tokens->getNextMeaningfulToken($closeBraceIndex);
+
+            if (null !== $afterCloseBraceIndex && !$tokens[$afterCloseBraceIndex]->equalsAny([';', [T_CLOSE_TAG]])) {
+                continue;
+            }
+
+            // must have 2 arguments
+
+            // first argument must be a variable with possibly array indexing
+            // after that nothing meaningful should be till the next `,` or `)`
+            // if `)` than we cannot fix (it is a single argument call)
+
+            $firstArgumentStop = $this->getFirstArgumentEnd($tokens, $firstArgumentStart);
+
+            if (null === $firstArgumentStop) {
+                return;
+            }
+
+            // $firstArgumentStop index points to the `,` token
+
+            // second argument can be about anything, we only want to make sure there is not
```

Comment:
It can not be `array_push($a, ...$b)`

Reply:
thanks, added!

Reply:
Thanks! But the comment here should also be adjusted.

Reply:
I realized too late :(  , but updated here https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4790/commits/1734fcfe302b2c717d68ac83d981e38486805138#diff-10d10d8e0c00ad91c4ab62427c98ac3bR127